js/main.js

```diff
@@ -20,4 +20,88 @@ $entryForm.addEventListener('submit', function (event) {
   data.entries.unshift(singleData);
   $imagePlaceholder.setAttribute('src', entryImagePlaceholder);
   $entryForm.reset();
+
+  // const $renderEntry = renderEntry(singleData);
+  entriesList.prepend(renderEntry(singleData));
+  viewSwap('entries');
+  data.view = 'entries';
+});
+
+function renderEntry(entry) {
+  const $li = document.createElement('li');
+
+  const $row = document.createElement('div');
+  $row.setAttribute('class', 'row');
+  $li.appendChild($row);
+
+  const $columnHalfOne = document.createElement('div');
+  $columnHalfOne.setAttribute('class', 'column-half');
+  $row.appendChild($columnHalfOne);
+
+  const $columnHalfTwo = document.createElement('div');
+  $columnHalfTwo.setAttribute('class', 'column-half');
+  $row.appendChild($columnHalfTwo);
+
+  const $imgLine = document.createElement('img');
+  $imgLine.setAttribute('src', entry.url);
+  $columnHalfOne.appendChild($imgLine);
+
+  const $h2Element = document.createElement('h2');
+  $h2Element.textContent = entry.title;
+  $columnHalfTwo.appendChild($h2Element);
+
+  const $pElement = document.createElement('p');
+  $pElement.textContent = entry.description;
+  $columnHalfTwo.appendChild($pElement);
+
+  return $li;
+}
+
+// Add an anchor to the navbar to show the ”entries” view. You will need to
+// create an event handler function that utilizes the viewSwap function to show
+// the proper view.
+const noEntriesText = document.querySelector('.no-entries');
+function toggleNoEntries() {
+  if (data.entries.length === 0) {
+
+    noEntriesText.classList.remove('hidden');
+  }
+}
+
+function viewSwap(view) {
+  const entryForm = document.querySelector('div[data-view="entry-form"]');
+  const entries = document.querySelector('div[data-view="entries"]');
```

Comment:
I would move these selectors up to the top of the file, so that way you don't need to query for these elements every time this function runs, this would be inefficient, because you're forcing the document to search for both of these elements every time